website/docs/docs/stubs.md

```diff
@@ -51,14 +52,33 @@ For example, assuming you wanted to use the latest version of a particular contr
 ```sh
 https://<yourdomain>.pactflow.io/pacts/provider/:provider/consumer/:consumer/latest/stub
 ```
-
 ## Stub behaviour
 
 Pact contracts may define multiple overlapping requests - for example when there are provider states.
 
-Where multiple matching interactions are found, the interactions will be sorted by response status, and the first one will be returned. This may lead to non-deterministic behaviour.
+If the request matches any interactions, it will return the first response based on the order in the pact file.
+
+If the request does not match, it will return the errors from the interaction with the least number of mismatches, followed by the order in the Pact file.
+
+This matching behaviour may be [configured](#configuration) using additional headers sent with a request.
+
+**Example**
 
-_NOTE: Only versions 1 and 2 of the [pact specification](https://github.com/pact-foundation/pact-specification/) are currently fully supported. Pacts using the v3 format may be used, however, any matching features added in v3 will currently be ignored._
+Given a pact with two interactions with query `a=1` and `a=1&b=2&c=4`. If the stub server receives a request with query `a=1&b=2`, you get the error that query parameter b was not expected, instead of query parameter c is missing.
+
+The same principle applies to headers.
+
+## Configuration
+
+You can configure the behaviour of the stub service at runtime, using HTTP headers sent with the stub requests.
+
+| Header | Type | Description | Default |
+|--------|------|-------------|---------|
+| `pactflow-stub-cors` | boolean  | Automatically responds to OPTIONS requests and return default CORS headers. For more on CORs, refer to the [section](#cors) below. | `true` |
+| `pactflow-stub-cors-referer` | boolean | When set to `true`, sets the CORS origin value to the hostname of the referer URL. If set to `false`, or if there is no referer header, sets it to '*". | `false` |
+| `pactflow-stub-provider-state` | string | Provider state regular expression used to filter the responses. | n/a |
```

Comment:
@rholshausen what if you have an interaction with multiple provider states. Can you provide multiple values to this header?

Reply:
No

Reply:
But you can write a regex that matches more than one, i.e. `state (one|two)` will match `state one` or `state two`